batch_object_tagger.tf

```diff
@@ -39,9 +39,53 @@ data "aws_iam_policy_document" "pdm_object_tagger" {
 
     resources = [
       "${data.terraform_remote_state.common.outputs.config_bucket.arn}/*",
-      "${data.terraform_remote_state.common.outputs.published_bucket.arn}/*"
     ]
   }
+
+  statement {
+    sid    = "AllowDecryptConfigBucketObjects"
+    effect = "Allow"
+
+    actions = [
+      "kms:Decrypt",
+      "kms:DescribeKey",
+    ]
+
+    resources = [
+      "${data.terraform_remote_state.common.outputs.config_bucket.arn}/*",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "pdm_object_tagger_published_bucket" {
+  statement {
+    sid    = "AllowS3Tagging"
+    effect = "Allow"
+
+    actions = [
+      "s3:ListObjects",
+      "s3:GetObject",
+      "s3:*Tagging"
+    ]
+
+    resources = [
+      "${data.terraform_remote_state.common.outputs.published_bucket.arn}/*",
+    ]
+  }
+
+  statement {
+    sid    = "AllowDecryptPublishedBucketObjects"
+    effect = "Allow"
+
+    actions = [
+      "kms:Decrypt",
+      "kms:DescribeKey",
+    ]
+
+    resources = [
+      "${data.terraform_remote_state.common.outputs.published_bucket.arn}/*}",
+      ]
+  }
```

Comment:
it has no need to access the data here, just tag it, so it shouldn't need decrypt or getobject

Reply:
GetObject required in order to retrieve metadata: https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html